compiler/qsc/src/interpret/stateful.rs

```diff
@@ -231,10 +231,10 @@ impl Interpreter {
         };
 
         eval_expr(
```

Comment:
At this point, isn't `eval_entry` almost functionally identical to `run()`? I wonder if it's worth consolidating those to functions into one as well? Especially since the wasm layer even has its own "run with shots" implementation.

That being said I understand if you want to keep this PR scoped only to this fix.

Reply:
I think you are right, we are very close to eliminating this function in favor of using `run`... I'll see if this PR can accomodate it and if it gets too messy I can move that refactor to a follow up.

Reply:
Unfortunately it does end up touching more than I'd like for inclusion here, so I'll do a separate refactor.